configure.ac

```diff
@@ -51,8 +51,29 @@ AC_C_INLINE
 dnl ac_check_enable_debug will make sure AC_PROG_CC doesn't add a
 dnl bunch of debugging flags, so there's no ordering requirement
 dnl between this and AC_PROG_CC.
-AS_IF([test "${ax_enable_debug}" = "no"], [CFLAGS="${CFLAGS} -O3"])
-
+AS_IF([test "${ax_enable_debug}" = "no"], [
+             dnl Enable O3 optimization
+             CFLAGS="${CFLAGS} -O3 -fPIC"
+
+             dnl dead code elim
+             CFLAGS="${CFLAGS} -ffunction-sections -fdata-sections -Wl,--gc-sections"
```

Comment:
`-Wl,--gc-sections` should be in LDFLAGS, not CFLAGS.

Reply:
I tried that first, it doesn't work.